src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexBatchSink.java

```diff
@@ -522,6 +493,35 @@ private void prepareRunStorageBucket(BatchSinkContext context) throws Exception
     if (bucket == null) {
       GCPUtils.createBucket(storage, bucketName, config.getLocation(), cmekKeyName);
     }
+    Map<String, String> outputProperties = new HashMap<>(validatingOutputFormat.getOutputFormatConfiguration());
```

Comment:
Do you know why this set of logic is in `prepareRunStorageBucket`, this should be in prepareRun or have its own method

Reply:
I'm not sure, it was there to begin with. It seems to follow the same pattern as `prepareRunBigQueryDataset` method. I believe it's for adding/initiating output. I would refrain from adding it to `prepareRun` since it's got two child methods based on asset type (i.e., `prepareRunBigQueryDataset` and `prepareRunStorageBucket`) and this is asset specific logic. Why would it be its own method? I don't feel strongly either way, but doesn't look like it's in the wrong method either.

Reply:
I think this is defined the required hadoop/mapreduce configs for reading from GCS

Reply:
It might be "cleaner" to move into a separate method `getStorageBucketOutputProperties`